### DISCUSSION TITLE
Ability to specify default location of new files

### DISCUSSION THREAD

**Discussion by ImJohnMDaniel** [forcedotcom/cli#2370] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently, when you create a new file, metadata, flow, etc in a scratch org and then execute the `force:source:pull` command, the files automatically show up in the `<projectDirectory--default:true>/main/default` folder.  The concept of "default" doesn't always make sense and I would like control over it.

**What are you trying to do**
I want a way to adjust the default location directory pattern from "/main/default" to something else

**Describe the solution you'd like**
Probably the simplest approach would be to either add a new environment variable to override the default location or allow a new attribute in the pacakgeDirectory section of the sfdx-project.json to allow an override.

**Describe alternatives you've considered**
Right now there is no alternative.

**Additional context**

**Comment by mshanemc** [comment]
Another alternative: if you know what you're pulling, you can use `source:retrieve` with `--tracksource` (it now lets you specify the output location).
